snowflake/Makefile

```diff
@@ -7,9 +7,17 @@ all: bin
 bin:
```

Comment:
we need a `generate` Make target so that we can easily re-generate the mocks

see https://betterprogramming.pub/a-gomock-quick-start-guide-71bee4b3a6f1 for a reference

Reply:
thanks, addressed all the comments above.

I'm not confident on the generate gomock part, feeling I'm doing it in a strange way. Could you help take a look?